src/ln/channelmanager.rs

```diff
@@ -1675,6 +1712,11 @@ impl ChannelManager {
 					if let Err(e) = self.monitor.add_update_monitor(chan_monitor.get_funding_txo().unwrap(), chan_monitor) {
 						match e {
 							ChannelMonitorUpdateErr::PermanentFailure => {
+								// TODO: There may be some pending HTLCs that we intended to fail
```

Comment:
Hmm normally thanks to #198, forward HTLCs (and so backwards if they fail)  will get updated to ChannelMonitor into Channel::commitment_signed, but does with current add_update_monitor API could we know which monitors failed to update among remote/in-memory ones.. ?

Reply:
I'm not quite sure what your point is here, but I think #198 practically solves this issue. The ChannelMonitorUpdateErr API says that if your local update fails to commit (ie the in-memory update failed) you must return PermanentFailure irrespective of remote monitor(s). Thus, we should only ever get here if the in-memory monitor update succeeded so the latest-tx-broadcast should result in a fail_backwards once the HTLC-Timeout commits. This needs a test, though, obviously, once they're both merged.

Reply:
I noted it in Permanent failure's comment. Indeed need more tests on these things.